feat: MCP planning mode base

franklinnwren opened the pull request:
## Description

- **Issue**: Multiple Discord messages were sent during coordination because agents could execute MCP tools (like `discord_send_message`) even during coordination phases
- **Solution**: Distinguish between planning (coordination) and execution (final presentation) phases

## Implementation Details

### 1. Configuration Classes (`agent_config.py`)
Added new `CoordinationConfig` dataclass:
```python
@dataclass
class CoordinationConfig:
    enable_planning_mode: bool = False
    planning_mode_instruction: str = "During coordination, describe what you would do without actually executing actions..."
```

Integrated into `AgentConfig` class:
```python
coordination_config: CoordinationConfig = field(default_factory=CoordinationConfig)
```

### 2. Orchestrator Logic (`orchestrator.py`)
Modified `_stream_agent_execution` method to:
- **Detect coordination phase**: Check if `workflow_phase == "coordinating"`
- **Add planning instructions**: Inject planning mode instructions into agent system messages
- **Tool filtering**: During planning mode, agents don't get mcp tools; during execution they get all tools

### 3. YAML Configuration (`cli.py`)
Enhanced config parsing to support coordination settings:
```python
coordination_settings = kwargs.get("orchestrator", {}).get("coordination", {})
if coordination_settings:
    orchestrator_config.coordination_config = CoordinationConfig(
        enable_planning_mode=coordination_settings.get("enable_planning_mode", False),
        planning_mode_instruction=coordination_settings.get("planning_mode_instruction", "...")
    )
```

### 4. Configuration Examples
Created two example configurations:
- `three_agents_discord_mcp_planning_mode.yaml`: Full Discord MCP example with planning mode

Example YAML structure:
```yaml
orchestrator:
  coordination:
    enable_planning_mode: true
    planning_mode_instruction: |
      PLANNING MODE ACTIVE: You are currently in the coordination phase...
```

## Behavior Changes

### During Coordination Phase (Planning Mode Enabled)
- **Agents cannot execute**: MCP tools
- **Agents receive**: Special planning mode instructions in their system message
- **Focus**: Planning, analysis, and coordination discussion

### During Final Presentation Phase
- **Winning agent gets**: Full access to all tools including MCP
- **Implementation**: Agent executes the plan developed during coordination

## Type of change
- [ ] Bug fix (`fix:`) - Non-breaking change which fixes an issue
- [x] New feature (`feat:`) - Non-breaking change which adds functionality
- [ ] Breaking change (`breaking:`) - Fix or feature that would cause existing functionality to not work as expected
- [ ] Documentation (`docs:`) - Documentation updates
- [ ] Code refactoring (`refactor:`) - Code changes that neither fix a bug nor add a feature
- [ ] Tests (`test:`) - Adding missing tests or correcting existing tests
- [ ] Chore (`chore:`) - Maintenance tasks, dependency updates, etc.
- [ ] Performance improvement (`perf:`) - Code changes that improve performance
- [ ] Code style (`style:`) - Changes that do not affect the meaning of the code (formatting, missing semi-colons, etc.)
- [ ] CI/CD (`ci:`) - Changes to CI/CD configuration files and scripts

## Checklist
- [x] I have run pre-commit on my changed files and all checks pass
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## How to Test
Unit tests at `test_planning_mode.py`
### Test CLI Command
Setting `enable_planning_mode` to be `false` in `tools/planning/five_agents_discord_mcp_planning_mode.yaml`
and running
```
uv run python3 -m massgen.cli --config tools/planning/five_agents_discord_mcp_planning_mode.yaml "Send a message to server A channel B with the content 'Good afternoon'"
```
you will see multiple "Good afternoon" messages

After setting `enable_planning_mode` back to `true`
```
uv run python3 -m massgen.cli --config tools/planning/five_agents_discord_mcp_planning_mode.yaml "Send a message to server A channel B with the content 'Good afternoon'"
```
only sends one "Good afternoon" message during the final answer stage

similar applies for twitter and notion scripts

A local test:

```
uv run python3 -m massgen.cli --config tools/planning/five_agents_filesystem_mcp_planning_mode.yaml "Write a new line with random content to [your-file-location]"
```

When `enable_planning_mode` is `false`, you will see several new lines are added to your file during the orchestration stage. But if `enable_planning_mode` is `true`, you will only see new lines in the final presentation

## Additional context
TODO: Add a list of whitelist/blacklist of mcp_tool in the configuration. So we don't have to write them into the yaml file
TODO: Enable multi-turn support for the planning mode

ncrispino commented on massgen/configs/three_agents_discord_mcp_planning_mode.yaml:
```diff
@@ -0,0 +1,108 @@
+      - Message reading, sending, and management
+      - Channel and server information access
+
+      IMPORTANT: During coordination phase, you are in PLANNING MODE - describe your intended actions without executing them. Only execute tools during final presentation phase.
```
I feel like we should remove all this planning information in the system messages (in this config yaml and all others). We already have a separate planning mode instruction, so I think this may just be redundant?

ncrispino commented on massgen/configs/three_agents_discord_mcp_planning_mode.yaml:
```diff
@@ -0,0 +1,108 @@
+
+      Focus on planning, analysis, and coordination rather than execution.
+
+# Legacy Coordination Settings (for reference - these are now under orchestrator.coordination)
+coordination:
+  voting_enabled: true
+  max_rounds: 3
+  consensus_threshold: 0.6
+  timeout_seconds: 300
```
Get rid of these (in this and all yamls). I believe these are not supported anymore; some may be from v1.

ncrispino commented on massgen/configs/three_agents_discord_mcp_planning_mode.yaml:
Can you move this and the other configs to a special `massgen/configs/tools/planning` dir so it is more organized?

franklinnwren replied:
Sure

ncrispino commented:
Is there a config with planning that doesn't require extra MCP-specific API keys? This could be helpful for testing or for users who want to see how planning works.

franklinnwren commented:
> Is there a config with planning that doesn't require extra MCP-specific API keys? This could be helpful for testing or for users who want to see how planning works.

Just addressed the comments and updated the testing spec so we have a
